include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -2001,19 +2025,34 @@ __pattern_set_difference(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __e
                 ::std::forward<_ExecutionPolicy>(__exec)),
             __first1, __last1, __result, oneapi::dpl::__internal::__brick_copy<__hetero_tag<_BackendTag>>{});
     }
-
-    return __pattern_hetero_set_op(__tag, ::std::forward<_ExecutionPolicy>(__exec), __first1, __last1, __first2,
-                                   __last2, __result, __comp, unseq_backend::_DifferenceTag());
+    if (__par_backend_hetero::__can_set_op_write_from_set_b(__exec))
```

Comment:
In addition to satisfying this condition, I believe we must also confirm that the underlying value type is trivially copyable before taking the one-shot reduce-then-scan path. Do we do this anywhere?

Reply:
Good catch.  I don't think we do... I'll address this.